proc_square_refl.py

```diff
@@ -0,0 +1,237 @@
+                + ','.join(('start '+str(j[0])+' length '+str(diff(j)) for j in pulse_range))))   # If there is more than one section that goes above half max
+    # it assumes theres more than one pulse 
+    pulse_range = pulse_range[0,:]
+    fl.plot(abs(d['ch',0]['t':tuple(pulse_range)]), alpha=0.1, color='k',  #shades in the section of pulse range (above half max) for 
+            linewidth=10)                                                  #control 
+    refl_blip_ranges = abs(d['ch',1]).contiguous(lambda x:
+            x > 0.06*x.data.max()) 
+    print("before filter",refl_blip_ranges)
+    refl_blip_ranges = filter_range(refl_blip_ranges)                      # repeats the filter range but for the reflected signal                
+    refl_blip_ranges.sort(axis=0) # they are sorted by range size, not first/last
+    print("after filter",refl_blip_ranges)
+    assert refl_blip_ranges.shape[0] == 2, "seems to be more than two tuning blips "
+    for thisrange in refl_blip_ranges:
+        fl.plot(abs(d['ch',1]['t':tuple(thisrange)]), alpha=0.1, color='k',
+                linewidth=10)
+    # }}}
+    # {{{ apply a linear phase to find any remaining fine offset of the pulse,
+    #     and demodulate
+    f_shift = nddata(r_[-0.1e6:0.1e6:200j],'f_test')
+    # perform and store 200 frequency de-modulations of signal
+    test_array = d['ch',0] * exp(-1j*2*pi*f_shift*d.fromaxis('t'))
+    # performs frequency shift to control signal
+    fine_adj_frq = test_array.sum('t').run(abs).argmax('f_test').item()
+    # when modulating by same frequency of the waveform,
+    # abs(sum(waveform)) will be a maximum
+    center_frq += fine_adj_frq
+    print(("found center frequency at %0.5f MHz"%(center_frq/1e6)))
+    d.ft('t') #Fourier Transform into freq domain
+    d.setaxis('t', lambda x: x - fine_adj_frq) #apply shift to x axis
+    fl.next('frequency domain\n%s'%searchstr)
+    fl.plot(d['t':(None,40e6)], label='demod and sliced',
+            alpha=0.5)
+    # }}}
+    # {{{ I have modified the zeroth order phasing --
+    # just ignore for now -- JF (though I explain for
+    # myself)
+    #def zeroth_order_ph(d, plot_name=None):
+
+        #r'''determine the covariance of the datapoints
+        #in complex plane, and use to phase the
+        #zeroth-order even if the data is both negative
+        #and positive'''
+        #eigenValues, eigenVectors = eig(cov(c_[
+            #d.data.real,
+            #d.data.imag].T
+            #))
+        # next 3 lines from stackexchange -- sort by
+        # eigenvalue
+        #idx = eigenValues.argsort()[::-1]   
+        #eigenValues = eigenValues[idx]
+        #eigenVectors = eigenVectors[:,idx]
+        # determine the phase angle from direction of the
+        # largest principle axis
+        #ph0 = arctan2(eigenVectors[1,0],eigenVectors[0,0])
+        #if plot_name:
+            #eigenVectors *= (eigenValues.reshape(-1,2)*ones((2,1)))/eigenValues.max()*abs(d.data).max()
```

Comment:
don't commit the commented lines (below, as well)

Reply:
done

Reply:
issue with axes on line 144 with matching units